internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestBatch.java

```diff
@@ -626,6 +626,37 @@ public void testKafkaInputBatchJob() throws Exception {
     testRepush(storeName, validator);
   }
 
+  @Test(timeOut = TEST_TIMEOUT)
+  public void testKafkaInputBatchJobWithZstdCompression() throws Exception {
+    VPJValidator validator = (avroClient, vsonClient, metricsRepository) -> {
+      // test single get
+      for (int i = 1; i <= 100; i++) {
+        Assert.assertEquals(avroClient.get(Integer.toString(i)).get().toString(), "test_name_" + i);
+      }
+    };
+    String storeName = testBatchStore(
+        inputDir -> new KeyAndValueSchemas(writeSimpleAvroFileWithStringToStringSchema(inputDir)),
+        properties -> {
+          properties.setProperty(SEND_CONTROL_MESSAGES_DIRECTLY, String.valueOf(true));
```

Comment:
can we test both True/False for this case?

Reply:
Right, we should test with compression metric/KIF true-false combination.